### PR TITLE
use S3 related API by import BR

### PR DESCRIPTION
**Pull request by 3pointer**
### What problem does this PR solve? 
remove S3 related API in cdc. to concentrate all s3 API in the same repo(br for now).
 
### What is changed and how it works?
1. remove S3 API
2. import BR 

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `tidb-cdc/cdc-ansible`

### CODE REVIEW

**Comment by 3pointer** [comment]
this PR is WIP, until we fix the dependency issue.

cdc requires br for s3 api and br depend master version of TiDB.
but after https://github.com/pingcap/ticdc/pull/837/files merged, cdc master branch rely on release-4.0 version of TiDB. which is not compatible with master. so we have to choose
1. cherry-pick br changes to release-4.0(designed on 5.0).
2. change cdc dependency back to master TiDB.

**Review by leoppro — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -62,3 +65,5 @@ require (
 	honnef.co/go/tools v0.0.1-2020.1.5 // indirect
 	upper.io/db.v3 v3.7.1+incompatible
 )
+
+replace github.com/pingcap/br => github.com/3pointer/br v0.0.0-20200817122845-968a9b5582e8
```
Comment (amyangfei) [inline go.mod:69]: remember to remove the go mod replace after https://github.com/pingcap/br/pull/463 is merged

**Pull request merged**